internal/ignition/ignition.go

```diff
@@ -468,6 +474,66 @@ func (g *installerGenerator) addBootstrapKubeletIpIfRequired(log logrus.FieldLog
 	return envVars, nil
 }
 
+func (g *installerGenerator) applyManifestPatches(ctx context.Context) error {
+	log := logutil.FromContext(ctx, g.log)
+	manifestsOpenShiftDir := filepath.Join(g.workDir, "openshift")
+
+	err := filepath.Walk(manifestsOpenShiftDir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if filepath.Ext(info.Name()) != ".patch" {
+			return nil
+		}
+
+		manifestPatchPath := filepath.Join(manifestsOpenShiftDir, info.Name())
+		log.Debugf("Applying the following patch: %s", manifestPatchPath)
+		manifestPatch, err := os.ReadFile(manifestPatchPath)
+		if err != nil {
+			return errors.Wrapf(err, "failed to read manifest patch %s", manifestPatchPath)
+		}
+		log.Debugf("read the manifest at %s", manifestPatchPath)
+
+		manifestPath := filepath.Join(manifestsOpenShiftDir, strings.TrimSuffix(info.Name(), ".patch"))
```

Comment:
For patches you allow both `.yml` and `.yaml`, yet when it comes time to find the actual YAML to be patched - you insist the the extension will match the original file's extension. 

Should we offer some flexibility here - should we try to find both `.yml` and `.yaml` installer manifests that match our `<x>.y(a)ml.patch` ?

Reply:
> Should we offer some flexibility here - should we try to find both .yml and .yaml installer manifests that match our <x>.y(a)ml.patch ?

Yes, good point. Let me add this. thanks.

Reply:
I was looking at this and decided not to look for both `yaml` and `yml` but rather expect the user to use the same extension as the manifest that should be patched.

For example, for `cluster-scheduler-02-config.yml` then `cluster-scheduler-02-config.yml.patch(_something)` should be used. A couple of reasons why I think this is probable a better approach:

1. It keeps consistency across the names of the manifests that will be patched
2. It makes the code simpler and easier to maintain.

The downside is that it may not super obvious to users that this is the case and they may push a `.yaml.patch` to patch a `.yml` manifest. 

Another option is to only allow `.yml.patch` but I'm not 100% sure that OpenShift creates only `.yml` files and this would limit us in the future. If we can confirm that OCP only generates `.yml` files, then we may be able to go ahead with this strategy.

Reply:
>may not super obvious to users

We can solve that by failing an installation in which we don't find an installer manifest matching one of the user patches. It's a good thing anyway, to catch user typos in file names. Tell the user "Either remove this patch or we will not continue installation, because we can't find a file matching it". Bonus points if you implement a handy little "did you mean: <file name>" with some off-the-shelf text distance library

>but I'm not 100% sure that OpenShift creates only .yml files

Yeah I'm not willing to bet on that, let's not make this assumption

Reply:
I like the proposal. 

Considering the urgency of this fix, would you be ok with me implementing your latest suggestions in a follow-up PR?

Reply:
Sure, it's on you to create the ticket/follow up on it, I won't have it in my TODO